migration/migration/015-tracing-compression.sql

```diff
@@ -0,0 +1,99 @@
+        ELSE
+            _is_timescaledb_oss = (SELECT edition = 'apache' FROM timescaledb_information.license);
+        END IF;
+    END IF;
+
+    IF _is_timescaledb_installed THEN
+        IF _is_multinode THEN
+            --need to clear the search path while creating distributed
+            --hypertables because otherwise the datanodes don't find
+            --the right column types since type names are not schema
+            --qualified if in search path.
+            _saved_search_path := current_setting('search_path');
+            SET search_path = pg_temp;
+        END IF;
+
+        PERFORM public.set_chunk_time_interval(
+            '_ps_trace.span'::regclass,
+            '1 hour'::interval
+        );
+        PERFORM public.set_chunk_time_interval(
+            '_ps_trace.event'::regclass,
+            '1 hour'::interval
+        );
+        PERFORM public.set_chunk_time_interval(
+            '_ps_trace.link'::regclass,
+            '1 hour'::interval
+        );
+
+        IF _is_multinode THEN
+            execute format('SET search_path = %s', _saved_search_path);
+        END IF;
+
+        IF (NOT _is_timescaledb_oss) AND _is_compression_available THEN
+            -- drop any compressed chunks for the three tables we are modifying
+            PERFORM public.drop_chunks(
+                format('%I.%I', hypertable_schema, hypertable_schema)::regclass,
+                (max(range_end) + interval '1 minute'),
+                verbose=>true
+            )
+            FROM timescaledb_information.chunks
+            WHERE is_compressed = true
+            AND (hypertable_schema, hypertable_name) IN
+            (
+                ('_ps_trace', 'span'),
+                ('_ps_trace', 'link'),
+                ('_ps_trace', 'event')
+            )
+            GROUP BY hypertable_schema, hypertable_name
+            ORDER BY hypertable_name DESC -- span table last
+            ;
+
+            PERFORM public.remove_compression_policy('_ps_trace.span', if_exists=>true);
+            PERFORM public.remove_compression_policy('_ps_trace.event', if_exists=>true);
+            PERFORM public.remove_compression_policy('_ps_trace.link', if_exists=>true);
+
+            ALTER TABLE _ps_trace.span SET (timescaledb.compress=false);
+            ALTER TABLE _ps_trace.event SET (timescaledb.compress=false);
+            ALTER TABLE _ps_trace.link SET (timescaledb.compress=false);
+
+            ALTER TABLE _ps_trace.span SET (timescaledb.compress, timescaledb.compress_segmentby='trace_id', timescaledb.compress_orderby='span_id,start_time');
+            ALTER TABLE _ps_trace.event SET (timescaledb.compress, timescaledb.compress_segmentby='trace_id', timescaledb.compress_orderby='span_id,time');
+            ALTER TABLE _ps_trace.link SET (timescaledb.compress, timescaledb.compress_segmentby='trace_id', timescaledb.compress_orderby='span_id,span_start_time');
```

Comment:
I think these settings should change. The trace_id should go into the orderby. Is that right @niksajakovljevic ?

Reply:
Yes but we should also remove `segment_by` . But I suggest we make that a separate PR

Reply:
is 1 hour chunk size still too big?

Reply:
@niksajakovljevic do the compression settings look good now? I think since this migration hasn't been released, we can continue to add the trace_id changes to it when ready in another PR